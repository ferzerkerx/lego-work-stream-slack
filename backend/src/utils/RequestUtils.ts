import { Request } from 'express';
import { DateUtils } from './DateUtils';

export class RequestUtils {
  static dateParam(req: Request, paramName: string): Date {
    const paramValue: string = req.query[paramName];
    if (!paramValue) {
      return null;
    }
    return DateUtils.parseDate(paramValue);
  }

  static numberParam(req: Request, paramName: string): number {
    const paramValue: string = req.query[paramName];
    if (!paramValue) {
      return null;
    }
    const result = Number(req.query[paramName]);
    if (isNaN(result)) {
      return null;
    }
    return result;
  }

  static stringParam(req: Request, paramName: string): string {
    const paramValue: string = req.query[paramName];
    if (!paramValue || paramValue === '') {
      return null;
    }
    return paramValue;
  }

  static booleanParam(req: Request, paramName: string): boolean {
    const paramValue: string = req.query[paramName];
    return paramValue == 'true';
  }

  static arrayParam(req: Request, paramName: string): Array<string> {
    const paramValue: string = req.query[paramName];
    if (!paramValue || paramValue.length == 0) {
      return null;
    }
    return paramValue.split(',');
  }
}
