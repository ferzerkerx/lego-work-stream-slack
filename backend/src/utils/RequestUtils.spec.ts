import { RequestUtils } from './RequestUtils';

describe('RequestUtils', () => {
  function fakeRequest(paramName, value): any {
    const result = {
      query: {},
    };
    result.query[paramName] = value;
    return result;
  }

  test('dateParam', () => {
    const paramName = 'dateParam';
    const req = fakeRequest(paramName, '2018-10-10');
    const value = RequestUtils.dateParam(req, paramName);
    expect(value).toMatchSnapshot();
  });

  test('invalid dateParam', () => {
    const paramName = 'dateParam';
    const req = fakeRequest(paramName, '2018-10-10');
    const value = RequestUtils.dateParam(req, 'someOtherName');
    expect(value).toBeNull();
  });

  test('numberParam', () => {
    const paramName = 'numberParam';
    const req = fakeRequest(paramName, '02');
    const value = RequestUtils.numberParam(req, paramName);
    expect(value).toMatchSnapshot();
  });

  test('invalid numberParam', () => {
    const paramName = 'numberParam';
    const req = fakeRequest(paramName, '02');
    const value = RequestUtils.numberParam(req, 'someOtherName');
    expect(value).toBeNull();
  });

  test('stringParam', () => {
    const paramName = 'stringParam';
    const req = fakeRequest(paramName, 'someValue');
    const value = RequestUtils.stringParam(req, paramName);
    expect(value).toMatchSnapshot();
  });

  test('invalid stringParam', () => {
    const paramName = 'stringParam';
    const req = fakeRequest(paramName, 'someValue');
    const value = RequestUtils.stringParam(req, 'someOtherName');
    expect(value).toBeNull();
  });

  test('arrayParam', () => {
    const paramName = 'arrayParam';
    const req = fakeRequest(paramName, 'value1, value2');
    const value = RequestUtils.arrayParam(req, paramName);
    expect(value).toMatchSnapshot();
  });

  test('invalid arrayParam', () => {
    const paramName = 'arrayParam';
    const req = fakeRequest(paramName, 'value1, value2');
    const value = RequestUtils.arrayParam(req, 'someOtherName');
    expect(value).toBeNull();
  });

  test('booleanParam', () => {
    const paramName = 'booleanParam';
    const req = fakeRequest(paramName, 'true');
    const value = RequestUtils.arrayParam(req, paramName);
    expect(value).toMatchSnapshot();
  });

  test('invalid booleanParam', () => {
    const paramName = 'booleanParam';
    const req = fakeRequest(paramName, 'someValue');
    const value = RequestUtils.arrayParam(req, 'someOtherName');
    expect(value).toBeFalsy();
  });
});
