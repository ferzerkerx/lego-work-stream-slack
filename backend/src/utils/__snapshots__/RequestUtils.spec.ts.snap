// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestUtils arrayParam 1`] = `
Array [
  "value1",
  " value2",
]
`;

exports[`RequestUtils dateParam 1`] = `2018-10-10T00:00:00.000Z`;

exports[`RequestUtils numberParam 1`] = `2`;

exports[`RequestUtils stringParam 1`] = `"someValue"`;
